src/sqlfluff/core/templaters/dbt.py

```diff
@@ -199,12 +206,18 @@ def process(self, *, fname, in_str=None, config=None, formatter=None):
         )
 
         try:
-            return self._unsafe_process(fname, in_str, config)
+            processed_result = self._unsafe_process(fname, in_str, config)
+            # Reset the fail counter
+            self._sequential_fails = 0
+            return processed_result
         except DbtCompilationException as e:
+            # Increment the counter
+            self._sequential_fails += 1
```

Comment:
Why would compilation fail the first time but subsequently work?

Reply:
some compilation fails are fatal (the kind that initially inspired this), but I've realised some aren't. They are where the compilation fails in a file-specific way.